include/swift/AST/SubstitutionMap.h

```diff
@@ -58,6 +58,8 @@ class SubstitutionMap {
 
   void addSubstitution(CanType type, Type replacement);
 
+  void replaceSubstitution(CanType type, Type replacement);
```

Comment:
Please find a way of doing what you want without this operation, because soon SubstitutionMaps will have a different representation that will make this difficult to do. Maybe build a brand new SubstitutionMap?